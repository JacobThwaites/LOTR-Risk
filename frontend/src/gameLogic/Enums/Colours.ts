export enum Colour {
    GREEN = 'green',
    YELLOW = 'yellow',
    RED = 'red',
    BLACK = 'black',
}

