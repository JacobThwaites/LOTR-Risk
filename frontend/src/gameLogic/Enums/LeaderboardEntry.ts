export interface LeaderboardEntry {
    colour: string,
    areasControlled: number,
    totalUnits: number,
}
