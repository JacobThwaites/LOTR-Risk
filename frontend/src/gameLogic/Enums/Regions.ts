import { Region } from '../Models/Region';
import { Areas } from './Areas';

const eriadorAreas = [Areas.FORLINDON, Areas.MITHLOND, Areas.HARLINDON, Areas.TOWER_HILLS, Areas.EVENDIM_HILLS, Areas.LUNE_VALLEY, Areas.THE_SHIRE];
const arnorAreas = [Areas.FORODWAITH, Areas.ANGMAR, Areas.EASTERN_ANGMAR, Areas.BORDERLANDS, Areas.NORTH_DOWNS, Areas.RHUDAUR, Areas.FORNOST, Areas.WEATHER_HILLS, Areas.BUCKLAND, Areas.OLD_FOREST, Areas.SOUTH_DOWNS];
const rhunAreas = [Areas.NORTH_RHUN, Areas.WITHERED_HEATH, Areas.ESGAROTH, Areas.SOUTH_RHUN];
const mirkwoodAreas = [Areas.CARROCK, Areas.NORTH_MIRKWOOD, Areas.ANDUIN_VALLEY, Areas.EASTERN_MIRKWOOD, Areas.SOUTH_MIRKWOOD];
const rohanAreas = [Areas.EREGION, Areas.MINHIRIATH, Areas.DUNLAND, Areas.ENEDWAITH, Areas.FANGORN, Areas.WEST_ROHAN, Areas.GAP_OF_ROHAN];
const rhovanionAreas = [Areas.MORIA, Areas.GLADDEN_FIELDS, Areas.LORIEN, Areas.EMYN_MUIL, Areas.BROWNLANDS, Areas.RHUN_HILLS, Areas.THE_WOLD, Areas.DEAD_MARSHES];
const gondorAreas = [Areas.DRUWAITH_IAUR, Areas.ANDRAST, Areas.ANFALAS, Areas.VALE_OF_ERECH, Areas.LAMEDON, Areas.BELFALAS, Areas.LEBENNIN, Areas.MINAS_TIRITH, Areas.ITHILIEN, Areas.SOUTH_ITHILIEN];
const mordorAreas = [Areas.UDUN_VALE, Areas.MOUNT_DOOM, Areas.BARAD_DUR, Areas.MINAS_MORGUL, Areas.GORGOROTH, Areas.NURN];
const haradwaithAreas = [Areas.HARONDOR, Areas.UMBAR, Areas.DEEP_HARAD, Areas.HARAD, Areas.NEAR_HARAD, Areas.KHAND];

interface ListOfRegions { 
    [key: string]: Region
}

export const Regions: ListOfRegions = {
    ERIADOR: new Region('eriador', eriadorAreas, 3),
    ARNOR: new Region('arnor', arnorAreas, 7),
    RHUN: new Region('rhun', rhunAreas, 2),
    MIRKWOOD: new Region('mirkwood', mirkwoodAreas, 4),
    ROHAN: new Region('rohan', rohanAreas, 4),
    ROHVANION: new Region('rhovanion', rhovanionAreas, 4),
    GONDOR: new Region('gondor', gondorAreas, 7),
    MORDOR: new Region('mordor', mordorAreas, 2),
    HARADWAITH: new Region('haradwaith', haradwaithAreas, 2),
}