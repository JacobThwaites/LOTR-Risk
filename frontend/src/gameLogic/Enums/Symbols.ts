export enum Symbol {
    ARCHER = "ARCHER",
    CAVALRY = "CAVALRY",
    EAGLE = "EAGLE",
    WILD_CARD = "WILD_CARD"
}
