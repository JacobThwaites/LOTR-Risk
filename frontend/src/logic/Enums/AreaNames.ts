export enum AreaName {
    FORLINDON = 'Forlindon',
    MITHLOND = 'Mithlond',
    LUNE_VALLEY = 'LuneValley',
    EVENDIM_HILLS = 'EvendimHills',
    TOWER_HILLS = 'TowerHills',
    THE_SHIRE = 'TheShire',
    HARLINDON = 'Harlindon',
    FORODWAITH = 'Forodwaith',
    ANGMAR = 'Angmar',
    EASTERN_ANGMAR = 'EasternAngmar',
    BORDERLANDS = 'Borderlands',
    NORTH_DOWNS = 'NorthDowns',
    RHUDAUR = 'Rhudaur',
    FORNOST = 'Fornost',
    WEATHER_HILLS = 'WeatherHills',
    OLD_FOREST = 'OldForest',
    BUCKLAND = 'Buckland',
    SOUTH_DOWNS = 'SouthDowns',
    CARROCK = 'Carrock',
    NORTH_MIRKWOOD = 'NorthMirkwood',
    EASTERN_MIRKWOOD = 'EasternMirkwood',
    SOUTH_MIRKWOOD = 'SouthMirkwood',
    ANDUIN_VALLEY = 'AnduinValley',
    NORTH_RHUN = 'NorthRhun',
    WITHERED_HEATH = 'WitheredHeath',
    SOUTH_RHUN = 'SouthRhun',
    ESGAROTH = 'Esgaroth',
    MINHIRIATH = 'Minhiriath',
    EREGION = 'Eregion',
    DUNLAND = 'Dunland',
    ENEDWAITH = 'Enedwaith',
    FANGORN = 'Fangorn',
    WEST_ROHAN = 'WestRohan',
    GAP_OF_ROHAN = 'GapOfRohan',
    MORIA = 'Moria',
    GLADDEN_FIELDS = 'GladdenFields',
    LORIEN = 'Lorien',
    THE_WOLD = 'TheWold',
    EMYN_MUIL = 'EmynMuil',
    BROWNLANDS = 'BrownLands',
    RHUN_HILLS = 'RhunHills',
    DEAD_MARSHES = 'DeadMarshes',
    DRUWAITH_IAUR = 'DruwaithIaur', 
    ANDRAST = 'Andrast',
    ANFALAS = 'Anfalas',
    VALE_OF_ERECH = 'ValeOfErech',
    LAMEDON = 'Lamedon',
    BELFALAS = 'Belfalas',
    LEBENNIN = 'Lebennin',
    MINAS_TIRITH = 'MinasTirith',
    ITHILIEN = 'Ithilien',
    SOUTH_ITHILIEN = 'SouthIthilien',
    UDUN_VALE = 'UdunVale',
    MOUNT_DOOM = 'MountDoom',
    BARAD_DUR = 'BaradDur',
    MINAS_MORGUL = 'MinasMorgul',
    GORGOROTH = 'Gorgoroth',
    NURN = 'Nurn',
    HARONDOR = 'Harondor',
    UMBAR = 'Umbar',
    DEEP_HARAD = 'DeepHarad',
    HARAD = 'Harad',
    NEAR_HARAD = 'NearHarad',
    KHAND = 'Khand',
}