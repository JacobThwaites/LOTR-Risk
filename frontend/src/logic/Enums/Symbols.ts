export enum Symbol {
    ELVEN_ARCHER = 1,
    DARK_RIDER = 3,
    EAGLE = 5,
}
