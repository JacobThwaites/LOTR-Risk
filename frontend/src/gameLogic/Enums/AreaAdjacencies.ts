import { AreaName } from './AreaNames';

interface IAdjacencies {
    [key: string]: Array<AreaName>
}

export const Adjacencies: IAdjacencies = {
}

Adjacencies[AreaName.FORLINDON] = [AreaName.MITHLOND];
Adjacencies[AreaName.MITHLOND] = [AreaName.FORLINDON, AreaName.LUNE_VALLEY, AreaName.HARLINDON, AreaName.TOWER_HILLS];
Adjacencies[AreaName.HARLINDON] = [AreaName.MITHLOND];
Adjacencies[AreaName.LUNE_VALLEY] = [AreaName.MITHLOND, AreaName.EVENDIM_HILLS, AreaName.BORDERLANDS, AreaName.TOWER_HILLS];
Adjacencies[AreaName.EVENDIM_HILLS] = [AreaName.LUNE_VALLEY, AreaName.TOWER_HILLS, AreaName.BORDERLANDS, AreaName.NORTH_DOWNS];
Adjacencies[AreaName.TOWER_HILLS] = [AreaName.MITHLOND, AreaName.EVENDIM_HILLS, AreaName.THE_SHIRE, AreaName.LUNE_VALLEY];
Adjacencies[AreaName.THE_SHIRE] = [AreaName.TOWER_HILLS, AreaName.BUCKLAND];
Adjacencies[AreaName.FORODWAITH] = [AreaName.BORDERLANDS, AreaName.ANGMAR, AreaName.EASTERN_ANGMAR, AreaName.NORTH_RHUN, AreaName.WITHERED_HEATH];
Adjacencies[AreaName.BORDERLANDS] = [AreaName.LUNE_VALLEY, AreaName.EVENDIM_HILLS, AreaName.NORTH_DOWNS, AreaName.ANGMAR, AreaName.FORODWAITH, AreaName.WEATHER_HILLS, AreaName.FORNOST];
Adjacencies[AreaName.ANGMAR] = [AreaName.FORODWAITH, AreaName.BORDERLANDS];
Adjacencies[AreaName.NORTH_DOWNS] = [AreaName.BORDERLANDS, AreaName.FORNOST];
Adjacencies[AreaName.FORNOST] = [AreaName.NORTH_DOWNS, AreaName.WEATHER_HILLS, AreaName.OLD_FOREST, AreaName.BUCKLAND, AreaName.BORDERLANDS];
Adjacencies[AreaName.WEATHER_HILLS] = [AreaName.FORNOST, AreaName.BORDERLANDS, AreaName.OLD_FOREST, AreaName.RHUDAUR, AreaName.SOUTH_DOWNS];
Adjacencies[AreaName.EASTERN_ANGMAR] = [AreaName.FORODWAITH, AreaName.CARROCK];
Adjacencies[AreaName.RHUDAUR] = [AreaName.WEATHER_HILLS, AreaName.CARROCK, AreaName.EREGION];
Adjacencies[AreaName.OLD_FOREST] = [AreaName.FORNOST, AreaName.WEATHER_HILLS, AreaName.BUCKLAND, AreaName.SOUTH_DOWNS];
Adjacencies[AreaName.BUCKLAND] = [AreaName.THE_SHIRE, AreaName.FORNOST, AreaName.OLD_FOREST, AreaName.SOUTH_DOWNS];
Adjacencies[AreaName.SOUTH_DOWNS] = [AreaName.BUCKLAND, AreaName.OLD_FOREST, AreaName.MINHIRIATH];
Adjacencies[AreaName.NORTH_RHUN] = [AreaName.FORODWAITH, AreaName.WITHERED_HEATH, AreaName.SOUTH_RHUN];
Adjacencies[AreaName.WITHERED_HEATH] = [AreaName.FORODWAITH, AreaName.NORTH_RHUN, AreaName.ESGAROTH];
Adjacencies[AreaName.ESGAROTH] = [AreaName.WITHERED_HEATH, AreaName.NORTH_MIRKWOOD];
Adjacencies[AreaName.MINHIRIATH] = [AreaName.SOUTH_DOWNS, AreaName.DUNLAND, AreaName.ENEDWAITH];
Adjacencies[AreaName.SOUTH_RHUN] = [AreaName.NORTH_RHUN, AreaName.BROWNLANDS];
Adjacencies[AreaName.CARROCK] = [AreaName.EASTERN_ANGMAR, AreaName.NORTH_MIRKWOOD, AreaName.ANDUIN_VALLEY, AreaName.RHUDAUR];
Adjacencies[AreaName.NORTH_MIRKWOOD] = [AreaName.CARROCK, AreaName.EASTERN_MIRKWOOD, AreaName.ESGAROTH];
Adjacencies[AreaName.ANDUIN_VALLEY] = [AreaName.CARROCK, AreaName.EASTERN_MIRKWOOD, AreaName.SOUTH_MIRKWOOD, AreaName.GLADDEN_FIELDS];
Adjacencies[AreaName.EASTERN_MIRKWOOD] = [AreaName.NORTH_MIRKWOOD, AreaName.ANDUIN_VALLEY, AreaName.SOUTH_MIRKWOOD, AreaName.BROWNLANDS];
Adjacencies[AreaName.SOUTH_MIRKWOOD] = [AreaName.ANDUIN_VALLEY, AreaName.EASTERN_MIRKWOOD, AreaName.BROWNLANDS, AreaName.EMYN_MUIL, AreaName.BROWNLANDS];
Adjacencies[AreaName.EREGION] = [AreaName.RHUDAUR, AreaName.MORIA, AreaName.DUNLAND];
Adjacencies[AreaName.DUNLAND] = [AreaName.EREGION, AreaName.MINHIRIATH, AreaName.ENEDWAITH];
Adjacencies[AreaName.ENEDWAITH] = [AreaName.MINHIRIATH, AreaName.DUNLAND, AreaName.GAP_OF_ROHAN];
Adjacencies[AreaName.WEST_ROHAN] = [AreaName.GAP_OF_ROHAN, AreaName.DRUWAITH_IAUR];
Adjacencies[AreaName.GAP_OF_ROHAN] = [AreaName.ENEDWAITH, AreaName.WEST_ROHAN, AreaName.FANGORN, AreaName.THE_WOLD, AreaName.MINAS_TIRITH];
Adjacencies[AreaName.FANGORN] = [AreaName.LORIEN, AreaName.THE_WOLD, AreaName.GAP_OF_ROHAN];
Adjacencies[AreaName.MORIA] = [AreaName.EREGION, AreaName.GLADDEN_FIELDS];
Adjacencies[AreaName.GLADDEN_FIELDS] = [AreaName.MORIA, AreaName.ANDUIN_VALLEY, AreaName.LORIEN];
Adjacencies[AreaName.LORIEN] = [AreaName.GLADDEN_FIELDS, AreaName.FANGORN, AreaName.THE_WOLD];
Adjacencies[AreaName.THE_WOLD] = [AreaName.LORIEN, AreaName.FANGORN, AreaName.EMYN_MUIL, AreaName.GAP_OF_ROHAN];
Adjacencies[AreaName.EMYN_MUIL] = [AreaName.ANDUIN_VALLEY, AreaName.SOUTH_MIRKWOOD, AreaName.BROWNLANDS, AreaName.DEAD_MARSHES, AreaName.THE_WOLD];
Adjacencies[AreaName.BROWNLANDS] = [AreaName.SOUTH_MIRKWOOD, AreaName.SOUTH_RHUN, AreaName.EMYN_MUIL, AreaName.RHUN_HILLS, AreaName.DEAD_MARSHES];
Adjacencies[AreaName.RHUN_HILLS] = [AreaName.BROWNLANDS];
Adjacencies[AreaName.DEAD_MARSHES] = [AreaName.EMYN_MUIL, AreaName.BROWNLANDS, AreaName.UDUN_VALE, AreaName.ITHILIEN];
Adjacencies[AreaName.DRUWAITH_IAUR] = [AreaName.WEST_ROHAN, AreaName.ANFALAS];
Adjacencies[AreaName.ANDRAST] = [AreaName.ANFALAS];
Adjacencies[AreaName.ANFALAS] = [AreaName.ANDRAST, AreaName.DRUWAITH_IAUR, AreaName.VALE_OF_ERECH];
Adjacencies[AreaName.VALE_OF_ERECH] = [AreaName.ANFALAS, AreaName.LAMEDON];
Adjacencies[AreaName.LAMEDON] = [AreaName.VALE_OF_ERECH, AreaName.BELFALAS, AreaName.LEBENNIN];
Adjacencies[AreaName.BELFALAS] = [AreaName.LAMEDON, AreaName.LEBENNIN];
Adjacencies[AreaName.LEBENNIN] = [AreaName.LAMEDON, AreaName.BELFALAS, AreaName.MINAS_TIRITH];
Adjacencies[AreaName.MINAS_TIRITH] = [AreaName.GAP_OF_ROHAN, AreaName.LEBENNIN, AreaName.ITHILIEN];
Adjacencies[AreaName.ITHILIEN] = [AreaName.DEAD_MARSHES, AreaName.SOUTH_ITHILIEN, AreaName.MINAS_MORGUL, AreaName.MINAS_TIRITH];
Adjacencies[AreaName.SOUTH_ITHILIEN] = [AreaName.ITHILIEN, AreaName.HARONDOR];
Adjacencies[AreaName.UDUN_VALE] = [AreaName.DEAD_MARSHES, AreaName.MOUNT_DOOM];
Adjacencies[AreaName.MOUNT_DOOM] = [AreaName.UDUN_VALE, AreaName.BARAD_DUR, AreaName.GORGOROTH];
Adjacencies[AreaName.BARAD_DUR] = [AreaName.MOUNT_DOOM, AreaName.GORGOROTH];
Adjacencies[AreaName.MINAS_MORGUL] = [AreaName.GORGOROTH, AreaName.ITHILIEN];
Adjacencies[AreaName.GORGOROTH] = [AreaName.MOUNT_DOOM, AreaName.BARAD_DUR, AreaName.MINAS_MORGUL, AreaName.NURN];
Adjacencies[AreaName.NURN] = [AreaName.GORGOROTH];
Adjacencies[AreaName.HARONDOR] = [AreaName.SOUTH_ITHILIEN, AreaName.HARAD];
Adjacencies[AreaName.UMBAR] = [AreaName.DEEP_HARAD];
Adjacencies[AreaName.DEEP_HARAD] = [AreaName.UMBAR, AreaName.HARAD];
Adjacencies[AreaName.HARAD] = [AreaName.DEEP_HARAD, AreaName.HARONDOR, AreaName.NEAR_HARAD];
Adjacencies[AreaName.NEAR_HARAD] = [AreaName.HARAD, AreaName.KHAND];
Adjacencies[AreaName.KHAND] = [AreaName.NEAR_HARAD];