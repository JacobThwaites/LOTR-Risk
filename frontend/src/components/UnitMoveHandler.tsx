import React, { Dispatch, SetStateAction } from "react";
import UnitInput from "./common/UnitInput";
import MoveUnitsButton from "./buttons/MoveUnitsButton";
import { AreaName } from "../gameLogic/Enums/AreaNames";
import areaDetails from "./svgPaths/AreaDetails";

type Props = {
  areaToMoveUnits: AreaName | null,
  unitsToMove: number,
  onMoveUnits: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void,
  setUnitsToMove: Dispatch<SetStateAction<number>>,
  isInputDisabled: boolean,
  label: string,
}

export default function UnitMoveHandler(props: Props) {
  let maxUnits = 0;
  
  if (props.areaToMoveUnits) {
    const areaDetail = areaDetails[props.areaToMoveUnits];
    maxUnits = areaDetail.units - 1;
  }

  return (
    <div className="unit-maneuver-handler">
      <div className="unit-maneuver-handler--input">
        <label>{props.label}</label>
        <UnitInput
          name="unitsToMove"
          max={maxUnits}
          onChange={(num: number) => props.setUnitsToMove(num)}
          value={props.unitsToMove}
          disabled={props.isInputDisabled}
        />
      </div>
      <MoveUnitsButton
        onMoveUnits={props.onMoveUnits}
        isDisabled={props.isInputDisabled}
      />
    </div>
  );
}