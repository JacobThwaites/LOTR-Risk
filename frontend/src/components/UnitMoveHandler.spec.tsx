import '@testing-library/jest-dom'
import React from 'react';
import { render } from '@testing-library/react'
import UnitMoveHandler from './UnitMoveHandler';
import { AreaName } from '../gameLogic/Enums/AreaNames';

it('renders UnitMoveHandler correctly', async () => {
    const areaToMoveUnits = AreaName.ANDRAST;
    const unitsToMove =  1;
    const onMoveUnits = () => {};
    const setUnitsToMove = () => {};
    const isInputDisabled = false;
    const label = 'label';

    const { container } = render(
        <UnitMoveHandler 
            areaToMoveUnits={areaToMoveUnits} 
            unitsToMove={unitsToMove} 
            onMoveUnits={onMoveUnits}
            setUnitsToMove={setUnitsToMove}
            isInputDisabled={isInputDisabled}
            label={label}
        />
    );

    expect(container).toMatchSnapshot();
});