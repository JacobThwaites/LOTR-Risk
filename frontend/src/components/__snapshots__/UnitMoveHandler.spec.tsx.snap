// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UnitMoveHandler correctly 1`] = `
<div>
  <div
    class="unit-maneuver-handler"
  >
    <div
      class="unit-maneuver-handler--input"
    >
      <label>
        label
      </label>
      <span
        class="react-numeric-input"
        style="position: relative; display: inline-block;"
      >
        <input
          class="number-selector"
          name="unitsToMove"
          pattern=".*"
          style="padding-right: 3ex; box-sizing: border-box; border: 1px solid #ccc; border-radius: 2px; padding-left: 4px; display: block; line-height: normal;"
          type="text"
          value="0"
        />
        <b
          style="position: absolute; right: 2px; width: 2.26ex; border-color: rgba(0,0,0,.1); border-style: solid; text-align: center; cursor: default; transition: all 0.1s; background: rgba(0, 0, 0, 0.1); box-shadow: -1px -1px 3px rgba(0,0,0,.1) inset,1px 1px 3px rgba(255,255,255,.7) inset; top: 2px; bottom: 50%; border-radius: 2px 2px 0 0; border-width: 1px 1px 0px 1px;"
        >
          <i
            style="position: absolute; top: 50%; left: 50%; width: 0px; height: 0px; border-width: 0px 0.6ex 0.6ex 0.6ex; border-color: transparent transparent rgba(0, 0, 0, 0.7); border-style: solid; margin: -0.3ex 0px 0px -0.56ex;"
          />
        </b>
        <b
          style="position: absolute; right: 2px; width: 2.26ex; border-color: rgba(0,0,0,.1); border-style: solid; text-align: center; cursor: default; transition: all 0.1s; background: rgba(0, 0, 0, 0.1); box-shadow: -1px -1px 3px rgba(0,0,0,.1) inset,1px 1px 3px rgba(255,255,255,.7) inset; top: 50%; bottom: 2px; border-radius: 0 0 2px 2px; border-width: 0px 1px 1px 1px;"
        >
          <i
            style="position: absolute; top: 50%; left: 50%; width: 0px; height: 0px; border-width: 0.6ex 0.6ex 0px 0.6ex; border-color: rgba(0, 0, 0, 0.7) transparent transparent; border-style: solid; margin: -0.3ex 0px 0px -0.56ex;"
          />
        </b>
      </span>
    </div>
    <button
      class="custom-button"
    >
      Move Units
    </button>
  </div>
</div>
`;
